apps/prairielearn/src/pages/instructorInstanceAdminPublishing/components/CourseInstancePublishingForm.tsx

```diff
@@ -0,0 +1,572 @@
+        ) {
+          updatedStartDate = nowTemporal;
+        }
+
+        if (originalEndDate && now <= originalEndDate) {
+          // Try to re-use the original end date if it is in the future.
+          updatedEndDate = dateToPlainDateTime(originalEndDate, courseInstance.display_timezone);
+        } else if (
+          currentEndDate === null ||
+          // If the current end date is in the past, set it to 18 weeks from now.
+          Temporal.PlainDateTime.compare(currentEndDate, nowTemporal) < 0
+        ) {
+          updatedEndDate = eighteenWeeksLater;
+        }
+        break;
+      }
+    }
+    setValue('start_date', updatedStartDate?.toString() ?? '');
+    setValue('end_date', updatedEndDate?.toString() ?? '');
+  };
+
+  const handleAddWeek = async (field: 'start_date' | 'end_date') => {
+    const currentValue = field === 'start_date' ? startDate : endDate;
+    if (currentValue) {
+      const currentDate = Temporal.PlainDateTime.from(currentValue);
+      const newValue = currentDate.add({ weeks: 1 });
+      setValue(field, newValue.toString());
+      // setValue doesn't seem to trigger validation so we need to trigger it manually
+      await trigger('start_date');
+      await trigger('end_date');
+    }
+  };
+
+  // Validation
+  const validateStartDate = (value: string) => {
+    if (selectedStatus === 'publish_scheduled') {
+      if (!value) {
+        return 'Start date is required for scheduled publishing';
+      }
+      // Check if start date is in the future
+      const startDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+      if (startDateTime <= nowRoundedToSeconds()) {
+        return 'Start date must be in the future for scheduled publishing';
+      }
+    }
+    return true;
+  };
+
+  const validateEndDate = (value: string) => {
+    if (selectedStatus !== 'unpublished') {
+      if (!value) {
+        return 'End date is required';
+      }
+      // Check if end date is after start date
+      if (startDate && value) {
+        const startDateTime = plainDateTimeStringToDate(startDate, courseInstance.display_timezone);
+        const endDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+        if (endDateTime <= startDateTime) {
+          return 'End date must be after start date';
+        }
+      }
+    }
+    return true;
+  };
+
+  return (
+    <>
+      <div class="mb-4">
+        <h4 class="mb-4">Publishing</h4>
+
+        {!canEdit && origHash !== null && (
+          <div class="alert alert-info" role="alert">
+            You do not have permission to edit access control settings.
+          </div>
+        )}
+        {!canEdit && origHash === null && (
+          <div class="alert alert-warning" role="alert">
+            You cannot edit access control settings because the <code>infoCourseInstance.json</code>{' '}
+            file does not exist.
+          </div>
+        )}
```

Comment:
I raised a discussion about naming on Slack. Given that this page talks about "publishing", I currently think we should not call this "access control". But let's discuss more on Slack.